src/main/java/com/iexec/core/chain/DealWatcherService.java

```diff
@@ -90,7 +94,38 @@ public DealWatcherService(ChainConfig chainConfig,
     @Async
     @EventListener(ApplicationReadyEvent.class)
     public void run() {
-        subscribeToDealEventFromOneBlockToLatest(configurationService.getLastSeenBlockWithDeal());
+        outOfService = false;
+        disposeSubscription(dealEventsSubscription);
+        dealEventsSubscription = subscribeToDealEventFromOneBlockToLatest(configurationService.getLastSeenBlockWithDeal());
+    }
+
+    /**
+     * Create new deal watching subscription.
+     */
+    @Async
+    @EventListener(ChainConnectedEvent.class)
+    public void reconnect() {
+        run();
+    }
```

Comment:
Is there a reason why you have 2 methods doing the exact same operations? Shouldn't we have a single `connect` method that would be triggered by both events?

Reply:
I just did not know it was possible
see 453ab8d